forum/models/model_utils.py

```diff
@@ -695,3 +695,74 @@ def get_username_from_id(user_id: str) -> Optional[str]:
     if username := user.get("username"):
         return username
     return None
+
+
+def find_subscribed_threads(user_id: str, course_id: str) -> list[str]:
+    """
+    Find threads that a user is subscribed to in a specific course.
+
+    Args:
+        user_id (str): The ID of the user.
+        course_id (str): The ID of the course.
+
+    Returns:
+        list: A list of thread ids that the user is subscribed to in the course.
+    """
+    subscriptions = Subscriptions()
+    threads = CommentThread()
+
+    subscription_filter = {"subscriber_id": user_id}
+    subscriptions_cursor = subscriptions.find(subscription_filter)
+
+    thread_ids = []
+    for subscription in subscriptions_cursor:
+        thread_ids.append(ObjectId(subscription["source_id"]))
+
+    thread_filter = {"_id": {"$in": thread_ids}, "course_id": course_id}
+    threads_cursor = threads.find(thread_filter)
+
+    subscribed_ids = []
+    for thread in threads_cursor:
+        subscribed_ids.append(thread["_id"])
+
+    return subscribed_ids
+
+
+def get_group_ids_from_params(params: dict[str, Any]) -> list[int]:
+    """
+    Extract group IDs from the provided parameters.
+
+    Args:
+        params (dict): A dictionary containing the parameters.
+
+    Returns:
+        list: A list of group IDs.
+
+    Raises:
+        ValueError: If both `group_id` and `group_ids` are specified in the parameters.
+    """
+
+    if "group_id" in params and "group_ids" in params:
+        raise ValueError("Cannot specify both group_id and group_ids")
+    group_ids = []
+    if "group_id" in params:
+        group_ids.append(int(params["group_id"]))
+    elif "group_ids" in params:
+        group_ids.extend([int(x) for x in params["group_ids"].split(",")])
+    return group_ids
+
+
+def subscribe_user(
+    user_id: str, source_id: str, source_type: str
+) -> dict[str, Any] | None:
+    """Subscribe a user to a source."""
+    subscription = Subscriptions().get_subscription(user_id, source_id)
+    if not subscription:
+        Subscriptions().insert(user_id, source_id, source_type)
+        subscription = Subscriptions().get_subscription(user_id, source_id)
```

Comment:
can't we use get method from MongoBaseModel, 
```
subscription_id = Subscriptions().insert(user_id, source_id, source_type)
subscription = Subscriptions().get(subscription_id)
```

Reply:
following the same pattern in tests as the api where we have `subscriber_id` and `source_id`

Reply:
yes, that will be also fine.